common/components/common/upload/ImageCropUploadButton.jsx

```diff
@@ -0,0 +1,188 @@
+// @flow
+
+import React from 'react';
+import type { S3Data } from './S3Data.jsx'
+import type {FileUploadData} from "./FileUploadButton";
+import FileSelectButton from './FileSelectButton.jsx';
+import ReactCrop from "react-image-crop";
+
+type Props = {|
+  onFileUpload: (FileUploadData) => void,
+  buttonText: string,
+  cropButtonText: string,
+  acceptedFileTypes: string,
+  iconClass: string,
+  aspect: number
+|};
+
+type State = {|
+  s3Key: string,
+|};
+
+class ImageCropUploadButton extends React.PureComponent<Props, State> {
+
+  constructor(props): void {
+    super(props);
+    const aspect = props.aspect || 1 / 1;
+    this.state = {
+      s3Key: "",
+      src: null,
+      isCropping: false,
+      croppedImageUrl: '',
+      crop: {
+        unit:"%",
+        width:30,
+        aspect: aspect
+      }
+    };
+  }
+
+  render(): React$Node {
+      const previewImage = this.state.croppedImageUrl || this.props.currentImage;
+      return (
+        <div className="ImageCropUploadButton-root">
+        {this.state.src && this.state.isCropping && (
+          <div className="ImageCropUploadButton-cropper">
+          <ReactCrop
+            src={this.state.src}
+            crop={this.state.crop}
+            onImageLoaded={this._onImageLoaded.bind(this)}
+            onComplete={this._onCropComplete.bind(this)}
+            onChange={this._onCropChange.bind(this)}
+          />
+            <input type="button" value={this.props.cropButtonText || "Done Cropping"} onClick={this._handleDoneCropping.bind(this)} className="ImageCropUploadButton-doneButton"/>
+</div>
+        )}
+        {!this.state.isCropping && (
+          <FileSelectButton hasImagePreview="true" 
+             previewImage={previewImage}
+             acceptedFileTypes="image/*"
+             dragText={this.props.dragText || "Drag Your Images Here or"}
```

Comment:
Do we support multiple upload for images? Maybe this should read image singular?